src/lib/asl.c

```diff
@@ -0,0 +1,56 @@
+#include <asl.h>
+#include <pwd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <time.h>
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+#include <xhyve/asl.h>
+
+static aslclient asl = NULL;
+static aslmsg log_msg = NULL;
+
+static unsigned char *buf = NULL;
+static size_t buf_size = 0;
+static size_t buf_capacity = 0;
+
+/* asl is deprecated in favor of os_log starting with macOS 10.12.  */
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+/* Grow buf/buf_capacity. */
+static void buf_grow(void)
+{
+	buf_capacity = buf_capacity ? 2 * buf_capacity : 1024;
```

Comment:
Why not initialise the buffer capacity to 1024 in `asl_init()` and then just grow from there? Also, we should probably add an upper bound unconditionally flush if it is reached. Or more generally, is there a reason you just don't allocate a fixed buffer of, say 4k and flush it when it's full (or on `\n`). Seem's much simpler.